extensions/ql-vscode/src/variant-analysis/variant-analysis-manager.ts

```diff
@@ -312,22 +307,35 @@ export class VariantAnalysisManager
     return problemQueries;
   }
 
-  private async runVariantAnalysisCommand(uri: Uri): Promise<void> {
-    // Build up details to pass to the functions that run the variant analysis.
-    const qlPackRootPath = await findPackRoot(uri.fsPath);
+  private async runVariantAnalysisCommand(queryFiles: Uri[]): Promise<void> {
+    if (queryFiles.length === 0) {
+      throw new Error("Please select a .ql file to run as a variant analysis");
+    }
+
+    const qlPackRootPath = await findPackRoot(queryFiles[0].fsPath);
```

Comment:
`findPackRoot` needs a little attention.

It works fine for the case where the queries are in a pack, but for the case where there is no pack it currently returns the immediate directory containing the query. For multiple queries not in a pack and not in the same directory this will make it complain they don't belong to the same query pack, when instead we could make it work by placing the pack root higher up the directory tree.

I suggest we change the output to `string | undefined` and then have a separate way of computing the ideal pack root in the case of multiple queries? That or `findPackRoot` takes an array of query paths and does this calculation and validation itself.

Reply:
Do you think it's worth changing the underlying code (run-remote-query.ts) so that it doesn't expect `qlPackRootPath` to always be set, instead of the change here? I've been tempted to change it but I thought that would cause too much churn and divert me from my task. But it does feel like it's the more correct thing to do. 

If we decide to go with that, I think it should be a separate PR though.

Reply:
As in make `QlPackDetails.qlPackRootPath` optional? That makes sense to me, and I was a little surprised when I read that interface to see that the field was always defined even if the query pack didn't exist.